app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php

```diff
@@ -223,8 +223,11 @@ public function getQuery(array $dimensions, string $productType, array $entityId
         $select->where("e.type_id = ?", $productType);
 
         if ($entityIds !== null) {
-            $select->where(sprintf('e.entity_id BETWEEN %s AND %s', min($entityIds), max($entityIds)));
-            $select->where('e.entity_id IN(?)', $entityIds);
+            if (count($entityIds) > 1) {
+                $select->where('e.entity_id IN (?)', $entityIds);
+            } else {
+                $select->where('e.entity_id = ?', $entityIds);
```

Comment:
It looks like just having IN (?) will be enough. I personally think it should work the same way.

Plus there was already a comment about it in previous pull request
https://github.com/magento/magento2/pull/22575#issuecomment-489086173
and this resource in the documentation says that it's automatically optimized.

Reply:
Hello @hws47a,
I have deleted if statement from my PR as it was in the previous PR, it was discussed with @okorshenko and decided that second `if` statement is not important in this case